solc/CommandLineInterface.cpp

```diff
@@ -789,19 +792,17 @@ bool CommandLineInterface::processInput()
 		bool successful = m_compiler->compile();
 
 		for (auto const& error: m_compiler->errors())
-			SourceReferenceFormatter::printExceptionInformation(
-				cerr,
+			formatter.printExceptionInformation(
 				*error,
-				(error->type() == Error::Type::Warning) ? "Warning" : "Error",
-				scannerFromSourceName
+				(error->type() == Error::Type::Warning) ? "Warning" : "Error"
```

Comment:
A next refactor could pull this expression into the formatter, I would say.

Reply:
Or actually just call `typeName()`.

Reply:
Yes, I did not want to rock the boat too much with this one, just lay the foundation for better encapsulation and configuration.